api/app/events/api.py

```diff
@@ -26,7 +26,12 @@ def event_info(user_id, event):
         'description': event.description,
         'start_date': event.start_date.strftime("%d %B %Y"),
         'end_date': event.end_date.strftime("%d %B %Y"),
-        'status': get_user_event_response_status(user_id, event.id)
+        'status': get_user_event_response_status(user_id, event.id),
+        'is_application_open': True if (datetime.now() < event.end_date and datetime.now() >= event.start_date) else False,
+        'is_review_open': True if (datetime.now() < event.review_close and datetime.now() >= event.review_open) else False,
+        'is_selection_open': True if (datetime.now() < event.selection_close and datetime.now() >= event.selection_open) else False,
+        'is_offer_open': True if (datetime.now() < event.offer_close and datetime.now() >= event.offer_open) else False,
+        'is_registration_open': True if (datetime.now() < event.registration_close and datetime.now() >= event.registration_open) else False
```

Comment:
Can we move this logic into the Event model? The reason is because I think these calculated fields may also be useful later down the line for future checks and I'd like to avoid the logic being duplicated (note we're not adding these fields to the db though). So the way I would imagine you could do it would be by using `@property`. The code would look like the following then

```
@property
def is_application_open(self):
    now = datetime.now()
    return now >= self.start_date and now < self.end_date
```
Then you will be able to call this calculated field like any other _actual_ field like so `event.is_application_open` (note no brackets)

To then pass these fields down to the front end, you can make the API return these fields like all the others, and the front end can just accept those fields as accurate without having to recalculate them again.

Reply:
This is another incredible piece of new information that I was never aware of. Thank you for taking the time to explain it to me.